### PR TITLE
[IMP] conf.py, *: interpolate the version and releases in the content

### PR DESCRIPTION
**Pull request by AntoineVDV**
This commit replaces hard-coded occurrences of the version and of the
last, current and next releases' version with placeholders interpolated
at build time to avoid manually updating these after each freeze.

task-2917614

### CODE REVIEW

File: content/developer/howtos/rdtraining/E_unittest.rst [hunk 1]
```diff
@@ -6,7 +6,7 @@ Advanced E: Python Unit Tests
 
 .. tip:: This tutorial assumes you followed the Core Training.
 
-  To do the exercise, fetch the branch 14.0-core from the repository XXX.
+  To do the exercise, fetch the branch {CURRENT_RELEASE}-core from the repository XXX.
```
Comment (Feyensv) [inline content/developer/howtos/rdtraining/E_unittest.rst:9]: Did anybody convert the training to latest versions ?

File: content/developer/reference/addons/testing.rst [hunk 2]
```diff
@@ -709,7 +709,7 @@ taken at the moment of the failure and written in
 
   '/tmp/odoo_tests/{db_name}/screenshots/'
 
-Two new command line arguments were added since Odoo 13.0 to control this behavior:
+Two new command line arguments were added since Odoo13.0 to control this behavior:
```
Comment (Feyensv) [inline content/developer/reference/addons/testing.rst:712]: ?

File: content/developer/howtos/rdtraining/E_unittest.rst (hunk 1 above)
Comment (AntoineVDV) [reply content/developer/howtos/rdtraining/E_unittest.rst:9]: Not yet. I guess creating the branches will be enough for now.

File: content/developer/reference/addons/testing.rst (hunk 2 above)
Comment (AntoineVDV) [reply content/developer/reference/addons/testing.rst:712]: I changed my mind but not the diff^^

File: content/administration/maintain/hosting_changes.rst [hunk 3]
```diff
@@ -33,15 +33,15 @@ From Odoo Online to on-premises
 ===============================
 
 1. Log into `your Odoo Online user portal <https://accounts.odoo.com/my/databases/manage>`_ and look for the version number of your database.
-2. If your database does not run a :ref:`major version <supported_versions>` of Odoo, you cannot host it on-premises yet, you have to upgrade it first to a new major version. (*e.g.: If your database runs Odoo 12.3 which is not a major version, you have to upgrade it first to Odoo 13.0 or 14.0.*)
+2. If your database does not run a :ref:`major version <supported_versions>` of Odoo, you cannot host it on-premises yet, you have to upgrade it first to a new major version. (*e.g.: If your database runs Odoo 12.3 which is not a major version, you have to upgrade it first to Odoo {CURRENT_RELEASE} or {CURRENT_RELEASE}.*)
```
Comment (Feyensv) [inline content/administration/maintain/hosting_changes.rst:36]: Hum, losing its meaning with your change i think

File: content/administration/maintain/hosting_changes.rst [hunk 4]
```diff
@@ -33,15 +33,15 @@ From Odoo Online to on-premises
...
 3. Download a backup of your database by clicking on the "Gear" icon next to your database name then :menuselection:`Download` (if the download fails due to your backup file being too large, contact `our support <https://www.odoo.com/help>`_)
 4. Restore it from the database manager on your local server.
 
 From Odoo Online to Odoo.sh
 ===========================
 
 1. Log into `your Odoo Online user portal <https://accounts.odoo.com/my/databases/manage>`_ and look for the version number of your database.
-2. If your database does not run a :ref:`major version <supported_versions>` of Odoo, you cannot host it on Odoo.sh yet, you have to upgrade it first to a new major version. (*e.g.: If your database runs Odoo 12.3 which is not a major version, you have to upgrade it first to Odoo 13.0 or 14.0.*)
+2. If your database does not run a :ref:`major version <supported_versions>` of Odoo, you cannot host it on Odoo.sh yet, you have to upgrade it first to a new major version. (*e.g.: If your database runs Odoo 12.3 which is not a major version, you have to upgrade it first to Odoo {CURRENT_RELEASE} or {CURRENT_RELEASE}.*)
```
Comment (Feyensv) [inline content/administration/maintain/hosting_changes.rst:44]: same

File: content/applications/websites/website/optimize/google_analytics.rst [hunk 5]
```diff
@@ -11,8 +11,8 @@ How to track your website's traffic in Google Analytics
   <https://support.google.com/analytics/answer/11583832>`_ are most likely still
   using **Universal Analytics**.
 
-  Odoo 15.0 is already adapted and ready for **Analytics 4**. After July 2023,
-  if you want to use Google Analytics tracking in versions before 15.0, you will
+  Odoo {LATEST_RELEASE} is already adapted and ready for **Analytics 4**. After July 2023,
+  if you want to use Google Analytics tracking in versions before {LATEST_RELEASE}, you will
```
Comment (Feyensv) [inline content/applications/websites/website/optimize/google_analytics.rst:15]: meaning change here

Comment (AntoineVDV) [reply content/applications/websites/website/optimize/google_analytics.rst:15]: That's intended. This page was updated in 14.0 to suggest upgrading to Odoo 15, but we should always suggest upgrading to whichever version was last released IMHO.

File: conf.py [hunk 6]
```diff
@@ -16,11 +16,23 @@
...
+# `current_release` is the version that was last released when the current branch was frozen. It has
+# the same value as `version` on release branches (XX.0) and has the value of the release branch's
+# `version` on SaaS branches and on master.
+# `previous_release` is the release before `current_release`.
+# `next_release` is the release coming after `current_release`.
+# `latest_release` is the version that was last released when the documentation was built. Similarly
+# to `current_release`, it does not take SaaS branches into account.
+current_release = '14.0'
+previous_release = '13.0'
+next_release = '15.0'
```
[the remark this answers is not in the document]
Comment (AntoineVDV) [reply conf.py]: > Also what happens for branches where the next release doesn't exist yet, is there still a next_release value with the future version or is there a synthetic value?

I'm actually considering removing that placeholder because I didn't find any usage for it in 14.0. WDYT @jcs-odoo?

If we keep it, I believe it would only be used to mention existing subsequent releases. For example, for stating that some feature is only available as of Odoo xx.0, which will would never write in the branch of the latest release or in master.

File: content/administration/maintain/update.rst [hunk 7]
```diff
@@ -19,7 +19,7 @@ definitions:
 
 Updating (an Odoo installation)
   Refers to the process of obtaining the latest revision of the source code for
-  your current Odoo Edition. For example, updating your Odoo Enterprise 13.0 to the
+  your current Odoo Edition. For example, updating your Odoo Enterprise {CURRENT_RELEASE} to the
   latest revision.
```
Comment (jcs-odoo) [inline content/administration/maintain/update.rst:23]: this makes less sense if current_release is also latest_release

```suggestion
  your current Odoo Edition. For example, updating Odoo Enterprise {PREVIOUS_RELEASE} to {CURRENT_RELEASE}, the
  latest revision.
```

File: conf.py [hunk 8]
```diff
@@ -16,11 +16,23 @@
...
+# the same value as `version` on release branches (XX.0) and has the value of the release branch's
+# `version` on SaaS branches and on master.
+# `previous_release` is the release before `current_release`.
+# `next_release` is the release coming after `current_release`.
+# `latest_release` is the version that was last released when the documentation was built. Similarly
+# to `current_release`, it does not take SaaS branches into account.
+current_release = '14.0'
+previous_release = str(float(current_release) - 1.0)
+next_release = str(float(current_release) + 1.0)
+latest_release = '16.0'
```
Comment (jcs-odoo) [inline conf.py:34]: How does this work with intermediary releases @AntoineVDV ?
if we're reading "Odoo Online" doc on 15.2, previous release isn't 14.2 nor is next release 16.2.

File: content/applications/websites/website/optimize/google_analytics.rst (hunk 5 above)
Comment (jcs-odoo) [reply content/applications/websites/website/optimize/google_analytics.rst:15]: I agree that we should push to upgrade to the latest release, but line 15 gives instructions that are version-specific. I would rephrase it.
"in older versions of Odoo" or something like that.

File: conf.py (hunk 8 above)
Comment (jcs-odoo) [reply conf.py:34]: @xmo-odoo Yes, there are intermediary versions of the doc: when you're on a 15.2 database and access the documentation from your database, you end up on the `saas-15.2` branch of the doc, displayed as "Odoo Online"
https://www.odoo.com/documentation/saas-15.2/

- 15.2 is not a major release, but it still is a release.
- These placeholders are about major releases --> current wording isn't clear about that.

File: content/administration/maintain/update.rst (hunk 7 above)
Comment (AntoineVDV) [reply content/administration/maintain/update.rst:23]: I reverted this change to leave 13.0.

Comment (AntoineVDV) [reply content/administration/maintain/update.rst:23]: Actually, I removed LATEST_RELEASE and NEXT_RELEASE following this comment and https://github.com/odoo/documentation/pull/3051#discussion_r1035914356

File: conf.py (hunk 8 above)
Comment (AntoineVDV) [reply conf.py]: I settled for `CURRENT_BRANCH` (saas-15.2), `CURRENT_VERSION` (15.2), `CURRENT_MAJOR_BRANCH` (15.0), `CURRENT_MAJOR_VERSION` (15).

File: conf.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (jcs-odoo) [reply conf.py]: True. It's best avoiding versions in the text whenever possible, anyway

File: conf.py (hunk 8 above)
Comment (jcs-odoo) [reply conf.py]: That's very clear like that 👍

**Comment by AntoineVDV** [comment]
@robodoo rebase-ff
